src/box/lua/experimental/connpool.lua

```diff
@@ -0,0 +1,59 @@
+local config = require('config')
+local checks = require('checks')
+local netbox = require('net.box')
+
+local connections = {}
+
+local function connect_using_config(instance_name, opts)
+    checks('string', {
+        wait_connected = '?boolean|number',
+        fetch_schema = '?boolean',
+    })
+    opts = opts or {}
+
+    local conn = connections[instance_name]
+    if conn == nil or conn.state == 'error' or conn.state == 'closed' or
+       conn.space == nil and opts.fetch_schema then
```

Comment:
I'm not sure about this check. First, `opts.fetch_schema` may be `nil` (and it is assumed as `true`). So, it should be `opts.fetch_schema ~= false` at least.

Second, it is possible that a space information is in fly. I guess that it is possible that space information fetching may be slower than retrying of the calls to `connect_using_config`. It means that a new connection will be created each time.

Reply:
Changed the condition.